### ISSUE TITLE
slower than the original monosdf

### ISSUE THREAD

**Issue by zjhthu** [autonomousvision/sdfstudio#13]
I found this implementation is slower than the original [monosdf](https://github.com/autonomousvision/monosdf). The CPU usage is quite high and GPU usage is about 30%.

So I
1. cache all images [here](https://github.com/autonomousvision/sdfstudio/blob/master/nerfstudio/data/datasets/base_dataset.py#L53)
2. implement a naive collate_fn without allocating memory [here](https://github.com/autonomousvision/sdfstudio/blob/master/nerfstudio/data/utils/nerfstudio_collate.py#L106) since batch size is 1 
3. [also transfer images to GPU ](https://github.com/autonomousvision/sdfstudio/blob/master/nerfstudio/data/utils/dataloaders.py#L188) to avoid [sampling in CPU](https://github.com/autonomousvision/sdfstudio/blob/master/nerfstudio/data/datamanagers/base_datamanager.py#L413)
4. remove the [evaluation code](https://github.com/autonomousvision/sdfstudio/blob/master/nerfstudio/engine/trainer.py#L174).

These approaches can speed up each iteration from ~800ms to ~350ms, but is still slower than the original monosdf. The CPU usage is still high, most of my 96 cores are used. The GPU usage is about 60%.
Do you have suggestions about this problem? 

Here is my test command:
`CUDA_VISIBLE_DEVICES=0,1,2,3 ns-train monosdf --pipeline.model.sdf-field.use-grid-feature True --pipeline.model.sdf-field.hidden-dim 128 --pipeline.model.sdf-field.num-layers 2 --pipeline.model.sdf-field.num-layers-color 2 --pipeline.model.sdf-field.use-appearance-embedding True --pipeline.model.sdf-field.geometric-init True --pipeline.model.sdf-field.inside-outside True  --pipeline.model.sdf-field.bias 0.6 --pipeline.model.sdf-field.beta-init 0.1 --pipeline.datamanager.train-num-images-to-sample-from 1 --pipeline.datamanager.train-num-times-to-repeat-images 0 --trainer.steps-per-eval-image 5000 --pipeline.model.background-model none --vis tensorboard --experiment-name monosdf-yungu-normal0.05-smooth --pipeline.model.mono-depth-loss-mult 0.001 --pipeline.model.mono-normal-loss-mult 0.05 --pipeline.model.smooth-loss-mult 0.005 --pipeline.datamanager.train-num-rays-per-batch 2048 --machine.num-gpus 4 --trainer.max-num-iterations 500 sdfstudio-data --data $MY_DATA --include_mono_prior True  --skip_every_for_val_split 30 `

**Comment by niujinshuchong** [comment]
Hi, the original monosdf use batch-size 1024 while you are using ` --pipeline.datamanager.train-num-rays-per-batch 2048`.

What gpus are you using? I also found the code is slower on some machines because of too many threads are used. You can get rid of this by adding ` OMP_NUM_THREADS=4` to your running command. I just tested without and with `OMP_NUM_THREADS=4` and here is the comparison
![image](https://user-images.githubusercontent.com/13434986/210569724-062b57fc-5a16-4dc1-adba-f9697d97ba1e.png)

**Comment by zjhthu** [comment]
My gpu is 2080ti.
Using fewer threads is helpful, 190 ms is used for each iteration! Thanks!

**Comment by zjhthu** [comment]
I am wondering which part of the code uses so many threads. I tried to set `num_workers` of `CacheDataloader` to 0 but it does not work.
